### DISCUSSION TITLE
[feedback question] why not use payment channels for storage deal payments

### DISCUSSION THREAD

**Discussion by ottozastrow** [filecoin-project/community#186] [General]
I am asking if there are reasons why this wouldn't be possible or helpful. 
If there are no blockers I would like to contribute code and design to this end.
currently a payment is made every 30 seconds to the storage miner. By using a payment channel between miner and customer the same payments can be made off chain. (The customer would probably want to make use of a watchtower to save effort)
Payment channels are already used for retrieval payments, but not for storage deals.
I would be happy to get some feedback!

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
the storage client signs a deal proposal and sends it to the miner, as soon as the miner publishes that proposal the client funds for the whole deal need to be locked in the market actor - there is no way back for the client to cancel the deal. the deal becomes "active" as soon as the miner commits the sealed sector to the chain. the market actor pays the miner over time. so it is not a payment from client to miner on the direct to ensure that the client has the funds available to pay the deal he proposes - the client signs an on-chain agreement.

**Reply by ottozastrow** [reply]
thanks a lot! I also found further clarification here
https://porcuquine.github.io/specs/docs/payments/
